labs-ios-starter/Surveys/View Controllers/TopicQuestionsViewController.swift

```diff
@@ -168,14 +161,92 @@ extension TopicQuestionsViewController: UICollectionViewDataSource, UICollection
     }
 
     func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
-        guard indexPath.row == questions.count else { return }
+        guard indexPath.row == questions?.count else { return }
 
         // questions+=1
         collectionView.reloadData()
     }
 }
 
-// MARK: - Segmented Control Delegate -
+extension TopicQuestionsViewController: UIPickerViewDelegate, UIPickerViewDataSource {
+    func numberOfComponents(in pickerView: UIPickerView) -> Int {
+        1
+    }
+
+    func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
+        questions?.count ?? 0
+    }
+
+    func pickerView(_ pickerView: UIPickerView, viewForRow row: Int, forComponent component: Int, reusing view: UIView?) -> UIView {
```

Comment:
wow this one is crazy long, maybe we could subclass this as a future goal to clean up the VC